drivers/sdhc/rcar_mmc.c

```diff
@@ -744,72 +744,81 @@ static int rcar_mmc_request(const struct device *dev,
 	struct sdhc_command *cmd,
 	struct sdhc_data *data)
 {
-	int ret;
+	int ret = -ENOTSUP;
 	uint32_t reg;
 	uint32_t response_type;
 	bool is_read = true;
+	int retries;
 
 	if (!dev || !cmd) {
 		return -EINVAL;
 	}
 
-	/* TODO: add retries mechanism, use retries counter from cmd for that */
 	response_type = cmd->response_type & SDHC_NATIVE_RESPONSE_MASK;
+	retries = cmd->retries ? : 1;
 
-	if (rcar_mmc_card_busy(dev)) {
-		LOG_ERR("%s: device busy", dev->name);
-		return -EBUSY;
-	}
+	while (ret && retries-- > 0) {
+		ret = rcar_mmc_poll_reg_flags_check_err(dev, RCAR_MMC_INFO2,
+							RCAR_MMC_INFO2_CBSY, 0, false, false);
+		if (ret) {
+			LOG_ERR("%s: device busy", dev->name);
```

Comment:
If upper layer configures number of retries, it expects that there may be an recoverable error. So I believe, you don't need to report each error with the `ERR` print. Other drivers use `LOG_DBG` in a such case.

Reply:
done

> Please provide a comprehensive commit description.

done